cli: add GEOS initialization to mt start-sql

otan opened the pull request:
Release note (bug fix): Fix a bug where multi-tenancy sql pods do
not start up correctly as it does not initialise the GEOS library
correctly.

Resolves #59254

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59259)
<!-- Reviewable:end -->

knz reviewed (approved):
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59259)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @sumeerbhola)



<!-- Sent from Reviewable.io -->

otan commented:
thanks!!!

bors r=knz

knz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59259)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @sumeerbhola)

___
*[pkg/cli/mt_start_sql.go, line 104 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59259#-MRaQqju6uk2AnYJDgOy:-MRaQqju6uk2AnYJDgOz:ben762r) ([raw file](https://github.com/cockroachdb/cockroach/blob/822fbd3e23e48f445ecd12d39fb7aaf4e9e68446/pkg/cli/mt_start_sql.go#L104)):*
> ```Go
> 	loc, err := geos.EnsureInit(geos.EnsureInitErrorDisplayPrivate, startCtx.geoLibsDir)
> 	if err != nil {
> 		log.Infof(ctx, "could not initialize GEOS - spatial functions may not be available: %v", err)
> ```

maybe `log.Ops.Warning` here
___
*[pkg/cli/mt_start_sql.go, line 106 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59259#-MRaQp2BCN2JTUwJnmit:-MRaQp2BCN2JTUwJnmiu:b-5vf4fn) ([raw file](https://github.com/cockroachdb/cockroach/blob/822fbd3e23e48f445ecd12d39fb7aaf4e9e68446/pkg/cli/mt_start_sql.go#L106)):*
> ```Go
> 		log.Infof(ctx, "could not initialize GEOS - spatial functions may not be available: %v", err)
> 	} else {
> 		log.Infof(ctx, "GEOS loaded from directory %s", loc)
> ```

I'd suggest `log.Ops.Infof` here

(same in the other file for regular servers)


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59259)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @sumeerbhola)

___
*[pkg/cli/mt_start_sql.go, line 104 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59259#-MRaQqju6uk2AnYJDgOy:-MRaRlQL77cUP7c0VjSk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/822fbd3e23e48f445ecd12d39fb7aaf4e9e68446/pkg/cli/mt_start_sql.go#L104)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

maybe `log.Ops.Warning` here
</blockquote></details>

Done.
___
*[pkg/cli/mt_start_sql.go, line 106 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59259#-MRaQp2BCN2JTUwJnmit:-MRaRlkJBMRg3hvFwry7:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/822fbd3e23e48f445ecd12d39fb7aaf4e9e68446/pkg/cli/mt_start_sql.go#L106)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I'd suggest `log.Ops.Infof` here

(same in the other file for regular servers)
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

knz commented:
thank you :100:

otan committed:
cli: add GEOS initialization to mt start-sql

Release note (bug fix): Fix a bug where multi-tenancy sql pods do
not start up correctly as it does not initialize the GEOS library
correctly.

otan commented:
bors r=knz
